core/src/unprocessed_packet_batches.rs

```diff
@@ -368,9 +374,40 @@ pub fn packet_message(packet: &Packet) -> Result<&[u8], DeserializedPacketError>
         .ok_or(DeserializedPacketError::SignatureOverflowed(sig_size))
 }
 
-/// Computes `(addition_fee + base_fee / requested_cu)` for `deserialized_packet`
-fn compute_fee_per_cu(_message: &VersionedMessage, _bank: &Bank) -> u64 {
-    1
+/// Computes `(addition_fee / requested_cu)` if feature tx_wide_compute_cap is enabled,
+/// otherwise return default 0u64
+fn compute_fee_per_cu(message: &VersionedMessage, bank: &Bank) -> u64 {
+    if bank.feature_set.is_active(&tx_wide_compute_cap::id()) {
```

Comment:
is this feature gate necessary?

Reply:
if the feature is not enabled, then it doesn't need to sanitize the message at all.

Reply:
Seems like we can parse the additional fee without sanitizing the messagefirst

All we need is the `message` from the `VersionedTransaction`: https://github.com/solana-labs/solana/blob/master/sdk/src/transaction/sanitized.rs#L97 and then parse the instructions here: https://github.com/solana-labs/solana/blob/master/program-runtime/src/compute_budget.rs#L112

Reply:
but should we parse without sanitizing first?

Reply:
you mean to simplify `fn sanitize_message(versioned_message)`  by doing: 
1. versioned_message.sanitize().ok()?; // still good to sanitize message before using it
2. straight process `message` from either legacy or V0 for compute_budget instruction? With some copied code?

Reply:
ok so after a helpful chat with @jstarry and @t-nelson you're right, you can't parse it out without the sanitize. Sorry for the confusion!

But this does mean we incur an extra account lookup for every transaction being added to the queue, potentially perf degrading